service/src/main/java/bio/terra/pipelines/service/PipelineInputsOutputsService.java

```diff
@@ -0,0 +1,199 @@
+package bio.terra.pipelines.service;
+
+import static bio.terra.pipelines.common.utils.FileUtils.constructDestinationBlobNameForUserInputFile;
+import static bio.terra.pipelines.common.utils.FileUtils.getBlobNameFromTerraWorkspaceStorageUrlGcp;
+
+import bio.terra.common.exception.InternalServerErrorException;
+import bio.terra.pipelines.common.utils.PipelineVariableTypesEnum;
+import bio.terra.pipelines.db.entities.Pipeline;
+import bio.terra.pipelines.db.entities.PipelineInput;
+import bio.terra.pipelines.db.entities.PipelineInputDefinition;
+import bio.terra.pipelines.db.entities.PipelineOutput;
+import bio.terra.pipelines.db.entities.PipelineOutputDefinition;
+import bio.terra.pipelines.db.entities.PipelineRun;
+import bio.terra.pipelines.db.repositories.PipelineInputsRepository;
+import bio.terra.pipelines.db.repositories.PipelineOutputsRepository;
+import bio.terra.pipelines.dependencies.gcs.GcsService;
+import bio.terra.pipelines.generated.model.ApiPipelineRunOutputs;
+import bio.terra.rawls.model.Entity;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/* Service to encapsulate the logic for processing pipeline inputs and outputs */
+@Service
+public class PipelineInputsOutputsService {
+  private final GcsService gcsService;
+
+  private final PipelineInputsRepository pipelineInputsRepository;
+  private final PipelineOutputsRepository pipelineOutputsRepository;
+  private final ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
so i think we want to autowire this in so we can use the one from the bean we create as opposed to the default one (we should do this everywhere we dont already do this)

Reply:
ah thanks, forgot about that